crates/miden-lib/asm/kernels/transaction/lib/memory.masm

```diff
@@ -1089,6 +1102,20 @@ export.set_num_storage_slots
     mem_store
 end
 
+#! Returns the type of the requested storage slot.
+#!
+#! Inputs:  [acct_storage_slots_section_ptr, index]
+#! Outputs: [slot_type]
+#!
+#! Where:
+#! - index is the location in memory of the storage slot.
+#! - slot_type is the type of the storage slot.
+export.get_storage_slot_type
```

Comment:
Why do we need to pass in `acct_storage_slots_section_ptr` into here? Can we infer slot type of the native account storage slot based just on the index?